cpp/oneapi/dal/graph/service_functions_test.cpp

```diff
@@ -0,0 +1,166 @@
+#include "oneapi/dal/test/engine/common.hpp"
+
+#include "oneapi/dal/graph/service_functions.hpp"
+#include "oneapi/dal/graph/undirected_adjacency_vector_graph.hpp"
+#include "oneapi/dal/io/graph_csv_data_source.hpp"
+#include "oneapi/dal/io/load_graph.hpp"
+
+namespace oneapi::dal::graph::test {
+
+namespace dal = oneapi::dal;
+
+using my_graph_type = dal::preview::undirected_adjacency_vector_graph<>;
+const dal::preview::load_graph::descriptor<dal::preview::edge_list<int32_t>, my_graph_type>
+    desc;
+
+auto get_data_source (const std::string& name) {
+	const std::string name_path = "cpp/oneapi/dal/graph/test_data/" + name;
+	const dal::preview::graph_csv_data_source ds(name_path);
+	return ds;
+}
+
+auto check_neighbors (const std::string& name, 
+	const dal::preview::vertex_size_type<my_graph_type> *correct_neigh,
```

Comment:
I suppose that shorthand `neigh` is not very common or well-established. I encourage you to use full word `neighbor` instead